src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -434,6 +440,34 @@ private void GenerateClassMarshals(Class @class)
             NewLine();
         }
 
+         private bool CheckClassIsStructible(Class @class)
```

Comment:
You have written this method for the second time. Please move it to some common to both generators location, such as ClassExtensions.